python/pyspark/rdd.py

```diff
@@ -188,6 +188,12 @@ def __init__(self, jrdd, ctx, jrdd_deserializer=AutoBatchedSerializer(PickleSeri
         self._id = jrdd.id()
         self.partitioner = None
 
+    def __enter__(self):
```

Comment:
We could do it - but users can still then do `with rdd.cache().map(...) as x:` and it would be valid. So it doesn't fully solve the issue.